presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1165,6 +1165,14 @@ public Node visitFunctionCall(SqlBaseParser.FunctionCallContext context)
 
         QualifiedName name = getQualifiedName(context.qualifiedName());
 
+        boolean ignoreNulls = false;
+        if (context.nullTreatment() != null) {
+            String functionName = name.toString();
+            check(functionName.equalsIgnoreCase("lag") || functionName.equalsIgnoreCase("lead"),
+                    "IGNORE NULLS clause not valid for '" + functionName + "' function", context);
+            ignoreNulls = context.nullTreatment().getText().equalsIgnoreCase("ignorenulls");
```

Comment:
This should use the `Token` type rather than string comparison. For example, see how window frame bounds are handled.